README.md
#Flutter GoHashLogin Package

  Flutter gohashlogin package help to make a login screen and Signup Screen


# Installation
```
1. Add the latest version of package to your pubspec.yaml (and run flutter pub get):
    dependencies:
        gohashlogin: ^0.0.3

2. Import the package and use it in your Flutter App.
    import 'package:gohashlogin/gohashlogin.dart';

```
# Output
![Screenshot from 2023-04-21 17-56-26](https://user-images.githubusercontent.com/99133851/233636302-89532a1f-6ed7-4b85-be0e-10437cab64b8.png)

![Screenshot from 2023-04-21 17-49-26](https://user-images.githubusercontent.com/99133851/233636392-97c56b7c-3940-422d-a911-4bd147083d4d.png)

## Usage
```
GoLogin(
        showAppBar: true,
        appbartitle: "Login",
        appBarTitleisCenter: true,
        useBackgroundImage: true,
        backgroundImage: "assets/background.jpg",
        textFormField: true,
        labelText: const [
          "Email",
          "Password",
        ],
        isObscure: true,
        sizedBox: const SizedBox(
          height: 200.0,
        ),
        decoration: [
          InputDecoration(
            labelText: "Email",
            hintText: "Enter Email",
            prefixIcon: const Align(
              widthFactor: 1.0,
              child: Icon(Icons.email),
            ),
            fillColor: Colors.white,
            focusedBorder: OutlineInputBorder(
              borderRadius: BorderRadius.circular(12),
              borderSide: const BorderSide(
                width: 1,
                style: BorderStyle.solid,
              ),
            ),
            border: OutlineInputBorder(
              borderRadius: BorderRadius.circular(12),
              borderSide: const BorderSide(
                width: 1,
                style: BorderStyle.solid,
              ),
            ),
            contentPadding: const EdgeInsets.all(16),
            filled: true,
          ),
        ],
        textField: 2,
        textController: [
          emailController,
          passwordController,
          // cpassController,
        ],
        showLoginButton: true,
        loginButtonTitle: "Login",
        buttonHeight: 60,
        buttonWidth: 150,
        ElevatedButton: ElevatedButton.styleFrom(
          elevation: 20,
          shape: StadiumBorder(),
          backgroundColor: Colors.blue,
        ),
        showForgotButton: true,
        onLoginPressed: () {},
        showAlreadyLogin: true,
        padding: EdgeInsets.only(bottom: 24),
        child: const Text.rich(
          TextSpan(
            text: "Don't have an Account?",
            style: TextStyle(fontSize: 20),
            children: [
              TextSpan(
                  text: ' Signup',
                  style: TextStyle(fontSize: 22, color: Colors.blue))
            ],
          ),
        ),
        onAlreadyLoginPressed: () {},
        onForgotPressed: () {},
      ),

```


## Additional information

TODO: Flutter package, which includes pre-built login and signup screens with form validation and background image, is a great example of the power of Flutter. By providing developers with a set of pre-built components, you are helping to speed up the development process and reduce the amount of time and effort required to create custom login and signup screens.





CHANGELOG.md
## 0.0.1

Modify Diffrent Login,Signup page Properties

## 0.0.2

Modify the Readme file and Example file

## 0.0.3

Modify the UI and add Some Functionality and add Example Tab 

test/gohashlogin_test.dart
// import 'package:flutter_test/flutter_test.dart';

// import 'package:gohashlogin/gohashlogin.dart';

// void main() {
//   test('adds one to input values', () {
//     final calculator = Calculator();
//     expect(calculator.addOne(2), 3);
//     expect(calculator.addOne(-7), -6);
//     expect(calculator.addOne(0), 1);
//   });
// }


lib/src/rendering.dart
part of gohashlogin;

// ignore: must_be_immutable
class GoLogin extends StatefulWidget {
  final bool showAppBar;
  final bool appBarTitleisCenter;
  String? appbartitle;
  final Color? statusBarColor;
  final Color? appBarColor;
  final Color? buttonTextColor;
  final bool useBackgroundImage;
  final bool useBackgroundColor;
  bool isObscure;
  bool isObscureCP;
  final String? backgroundImage;
  Widget? sizedBox;
  final bool textFormField;
  List<String>? labelText;
  List<String>? hintText;
  final List<IconData>? prefixIcon;
  final Color? fillColor;
  final bool filled;
  final int? textField;
  final List<TextEditingController>? textController;
  List? decoration;
  final VoidCallback onLoginPressed;
  final VoidCallback onAlreadyLoginPressed;
  final VoidCallback onForgotPressed;
  final bool showAlreadyLogin;
  Widget? child;
  final EdgeInsetsGeometry? padding;
  double? buttonWidth;
  double? buttonHeight;
  Widget? style;
  final ElevatedButton;
  final bool showLoginButton;
  String? loginButtonTitle;
  String? forgotButtonTitle;
  final bool showForgotButton;

  GoLogin({
    super.key,
    this.showAppBar = false,
    this.appBarTitleisCenter = false,
    this.appbartitle,
    this.statusBarColor,
    this.appBarColor,
    this.buttonTextColor,
    this.useBackgroundImage = false,
    this.useBackgroundColor = false,
    this.isObscure = false,
    this.isObscureCP = false,
    this.backgroundImage,
    this.sizedBox,
    this.textFormField = false,
    this.labelText,
    this.hintText,
    this.prefixIcon,
    this.textField,
    this.fillColor,
    this.filled = false,
    this.textController,
    this.decoration,
    required this.onLoginPressed,
    required this.onAlreadyLoginPressed,
    required this.onForgotPressed,
    this.showAlreadyLogin = false,
    this.child,
    this.padding,
    this.buttonWidth,
    this.buttonHeight,
    this.style,
    this.ElevatedButton,
    this.showLoginButton = false,
    this.loginButtonTitle,
    this.forgotButtonTitle,
    this.showForgotButton = false,
  });

  @override
  GoLoginState createState() => GoLoginState();
}

class GoLoginState extends State<GoLogin> {
  final _formKey = GlobalKey<FormState>();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: widget.showAppBar
          ? AppBar(
              backgroundColor: widget.appBarColor ?? Colors.blue,
              title: Text(widget.appbartitle ?? 'Example'),
              centerTitle: widget.appBarTitleisCenter,
            )
          : null,
      body: Container(
        decoration: BoxDecoration(
          color: widget.useBackgroundColor ? Colors.white : Colors.white,
          image: widget.useBackgroundImage && widget.backgroundImage != null
              ? DecorationImage(
                  image: AssetImage(widget.backgroundImage!),
                  fit: BoxFit.cover,
                )
              : null,
        ),
        child: Form(
          key: _formKey,
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              SizedBox(child: widget.sizedBox),
              if (widget.textFormField)
                ListView(
                  shrinkWrap: true,
                  children: [
                    ListView.builder(
                        shrinkWrap: true,
                        itemCount: widget.textField,
                        itemBuilder: (BuildContext context, int index) {
                          return Padding(
                            padding: const EdgeInsets.all(16.0),
                            child: TextFormField(
                              controller: widget.textController![index],
                              obscureText:
                                  widget.labelText![index] == "Password"
                                      ? widget.isObscure
                                      : widget.labelText![index] ==
                                              "Confirm Password"
                                          ? widget.isObscureCP
                                          : false,
                              decoration: widget.labelText![index] == "Password"
                                  ? InputDecoration(
                                      labelText: 'Password',
                                      hintText: 'Enter Password',
                                      prefixIcon: const Align(
                                        widthFactor: 1.0,
                                        child: Icon(
                                          Icons.lock,
                                        ),
                                      ),
                                      border: OutlineInputBorder(
                                        borderRadius: BorderRadius.circular(12),
                                        borderSide: const BorderSide(
                                          width: 1,
                                          style: BorderStyle.solid,
                                        ),
                                      ),
                                      filled: true,
                                      contentPadding: const EdgeInsets.all(16),
                                      fillColor: Colors.white,
                                      suffixIcon: IconButton(
                                          icon: Icon(widget.isObscure
                                              ? Icons.visibility
                                              : Icons.visibility_off),
                                          onPressed: () {
                                            setState(() {
                                              widget.isObscure =
                                                  !widget.isObscure;
                                            });
                                          }),
                                    )
                                  : widget.labelText![index] ==
                                          "Confirm Password"
                                      ? InputDecoration(
                                          labelText: 'Confirm Password',
                                          hintText: 'Enter Re Password',
                                          prefixIcon: const Align(
                                            widthFactor: 1.0,
                                            child: Icon(
                                              Icons.lock,
                                            ),
                                          ),
                                          border: OutlineInputBorder(
                                            borderRadius:
                                                BorderRadius.circular(12),
                                            borderSide: const BorderSide(
                                              width: 1,
                                              style: BorderStyle.solid,
                                            ),
                                          ),
                                          filled: true,
                                          contentPadding:
                                              const EdgeInsets.all(16),
                                          fillColor: Colors.white,
                                          suffixIcon: IconButton(
                                              icon: Icon(widget.isObscureCP
                                                  ? Icons.visibility
                                                  : Icons.visibility_off),
                                              onPressed: () {
                                                setState(() {
                                                  widget.isObscureCP =
                                                      !widget.isObscureCP;
                                                });
                                              }),
                                        )
                                      : widget.decoration![index],
                              keyboardType: TextInputType.emailAddress,
                              validator: (value) {
                                if (widget.labelText![index] == "Email") {
                                  if (value!.isEmpty) {
                                    return 'Please Enter Email';
                                  }
                                  if (!RegExp(
                                          "^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+.[a-z]")
                                      .hasMatch(value)) {
                                    return 'Please enter a valid Email';
                                  }
                                }

                                if (widget.labelText![index] == "User") {
                                  if (value!.isEmpty) {
                                    return "Please Enter UserName";
                                  }
                                  if (value.length < 2) {
                                    return "Please enter a valid Name";
                                  }
                                }

                                if (widget.labelText![index] == "Mobile") {
                                  if (value!.isEmpty) {
                                    return "Please Enter Mobile";
                                  }
                                  if (value.length < 9) {
                                    return "Please enter a valid Mobile Number";
                                  }
                                }
                                if (widget.labelText![index] == "DOB") {
                                  if (value!.isEmpty) {
                                    return "Please Enter Mobile";
                                  }
                                }

                                if (widget.labelText![index] == "Password") {
                                  if (value!.isEmpty) {
                                    return "Please Enter Password";
                                  }
                                  if (value.length < 8) {
                                    return "Password must be atleast 8 characters long";
                                  }
                                }
                                if (widget.labelText![index] ==
                                    "Confirm Password") {
                                  if (value!.isEmpty) {
                                    return "Please Re-Enter New Password";
                                  }
                                  if (value.length < 8) {
                                    return "Password must be atleast 8 characters long";
                                  }
                                  if (value !=
                                      widget
                                          .textController![
                                              widget.textField! - 2]
                                          .text) {
                                    return "Password must be same as above";
                                  }
                                }
                                return null;
                              },
                            ),
                          );
                        }),
                  ],
                ),
              if (widget.showForgotButton)
                Padding(
                  padding: const EdgeInsets.symmetric(horizontal: 8.0),
                  child: Row(
                    mainAxisAlignment: MainAxisAlignment.end,
                    children: [
                      TextButton(
                        onPressed: () {
                          widget.onForgotPressed();
                        },
                        child: Text(
                          widget.forgotButtonTitle ?? "Forgot Password?",
                          textAlign: TextAlign.end,
                        ),
                      ),
                    ],
                  ),
                ),
              const Spacer(),
              if (widget.showLoginButton)
                Container(
                  width: widget.buttonWidth ?? 150,
                  height: widget.buttonHeight ?? 60,
                  child: Padding(
                    padding: const EdgeInsets.only(bottom: 16.0),
                    child: ElevatedButton(
                      onPressed: () {
                        if (_formKey.currentState!.validate()) {
                          widget.onLoginPressed();
                        }
                      },
                      child: Text(
                        widget.loginButtonTitle ?? 'Login',
                      ),
                      style: widget.ElevatedButton ??
                          ElevatedButton.styleFrom(
                              elevation: 20,
                              shape: StadiumBorder(),
                              backgroundColor: Colors.blue),
                    ),
                  ),
                ),
              if (widget.showAlreadyLogin)
                Padding(
                  padding:
                      widget.padding ?? const EdgeInsets.only(bottom: 24.0),
                  child: InkWell(
                    onTap: () {
                      widget.onAlreadyLoginPressed();
                    },
                    child: widget.child ??
                        const Text.rich(
                          TextSpan(
                            text: "Already Have An Account ",
                            style: TextStyle(fontSize: 20),
                            children: [
                              TextSpan(
                                  text: ' Login',
                                  style: TextStyle(
                                      fontSize: 22, color: Colors.blue))
                            ],
                          ),
                        ),
                  ),
                ),
            ],
          ),
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:gohashlogin/gohashlogin.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        title: 'Flutter Demo',
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: const DataPage());
  }
}

class DataPage extends StatefulWidget {
  const DataPage({super.key});

  @override
  State<DataPage> createState() => _DataPageState();
}

class _DataPageState extends State<DataPage> {
  final TextEditingController emailController = TextEditingController();
  final TextEditingController passwordController = TextEditingController();
  final TextEditingController cpassController = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: GoLogin(
        showAppBar: true,
        appbartitle: "Login",
        appBarTitleisCenter: true,
        useBackgroundImage: true,
        backgroundImage: "assets/background.jpg",
        textFormField: true,
        labelText: const [
          "Email",
          "Password",
        ],
        isObscure: true,
        sizedBox: const SizedBox(
          height: 200.0,
        ),
        decoration: [
          InputDecoration(
            labelText: "Email",
            hintText: "Enter Email",
            prefixIcon: const Align(
              widthFactor: 1.0,
              child: Icon(Icons.email),
            ),
            fillColor: Colors.white,
            focusedBorder: OutlineInputBorder(
              borderRadius: BorderRadius.circular(12),
              borderSide: const BorderSide(
                width: 1,
                style: BorderStyle.solid,
              ),
            ),
            border: OutlineInputBorder(
              borderRadius: BorderRadius.circular(12),
              borderSide: const BorderSide(
                width: 1,
                style: BorderStyle.solid,
              ),
            ),
            contentPadding: const EdgeInsets.all(16),
            filled: true,
          ),
        ],
        textField: 2,
        textController: [
          emailController,
          passwordController,
          // cpassController,
        ],
        showLoginButton: true,
        loginButtonTitle: "Login",
        buttonHeight: 60,
        buttonWidth: 150,
        ElevatedButton: ElevatedButton.styleFrom(
          elevation: 20,
          shape: StadiumBorder(),
          backgroundColor: Colors.blue,
        ),
        showForgotButton: true,
        onLoginPressed: () {},
        showAlreadyLogin: true,
        padding: EdgeInsets.only(bottom: 24),
        child: const Text.rich(
          TextSpan(
            text: "Don't have an Account?",
            style: TextStyle(fontSize: 20),
            children: [
              TextSpan(
                  text: ' Signup',
                  style: TextStyle(fontSize: 22, color: Colors.blue))
            ],
          ),
        ),
        onAlreadyLoginPressed: () {},
        onForgotPressed: () {},
      ),
    );
  }
}
